jujugui/static/gui/src/app/components/user-profile/user-profile.js

```diff
@@ -58,16 +52,18 @@ YUI.add('user-profile', function() {
 
     componentWillMount: function() {
       this._fetchEnvironments();
-      if (this.props.authenticated) {
+      if (this.props.users.charmstore) {
         this._fetchEntities('charm');
         this._fetchEntities('bundle');
       }
     },
 
     componentDidUpdate: function(prevProps, prevState) {
-      // If the user has just been authenticated then update the data.
-      if (!prevProps.authenticated && this.props.authenticated) {
+      // If the user has changed then update the data.
+      if (prevProps.user !== this.props.user) {
         this._fetchEnvironments();
+      }
+      if (prevProps.users.charmstore !== this.props.users.charmstore) {
```

Comment:
Do we need to check for `prevProps.users` and `this.props.users` here? If they were not previously authenticated might they not have a .charmstore property?